plugins/opentelemetry/ir_ot_logger.go

```diff
@@ -0,0 +1,82 @@
+package opentelemetry
+
+import (
+	"fmt"
+
+	"github.com/blueprint-uservices/blueprint/blueprint/pkg/coreplugins/service"
+	"github.com/blueprint-uservices/blueprint/blueprint/pkg/ir"
+	"github.com/blueprint-uservices/blueprint/plugins/golang"
+	"github.com/blueprint-uservices/blueprint/plugins/golang/gocode"
+	"github.com/blueprint-uservices/blueprint/plugins/golang/goparser"
+	"github.com/blueprint-uservices/blueprint/plugins/workflow"
+	"golang.org/x/exp/slog"
+)
+
+// Blueprint IR Node that represents a process-level OT trace logger
+type OTTraceLogger struct {
+	golang.Node
+	golang.Instantiable
+
+	LoggerName  string
+	Iface       *goparser.ParsedInterface
+	Constructor *gocode.Constructor
+}
+
+func newOTTraceLogger(name string) (*OTTraceLogger, error) {
+	node := &OTTraceLogger{}
+	err := node.init(name)
+	if err != nil {
+		return nil, err
+	}
+	node.LoggerName = name
+	return node, nil
+}
+
+func (node *OTTraceLogger) init(name string) error {
+	workflow.Init("../../runtime")
+	spec, err := workflow.GetSpec()
+	if err != nil {
+		return err
+	}
+
+	details, err := spec.Get("OTTraceLogger")
+	if err != nil {
+		return err
+	}
+
+	node.Iface = details.Iface
+	node.Constructor = details.Constructor.AsConstructor()
+	return nil
+}
+
+func (node *OTTraceLogger) Name() string {
```

Comment:
Can you make sure to add a comment saying which interface each of these various functions implement.

Useful just because we have a lot of IR interfaces floating around.